internal/dataplane/parser/parser.go

```diff
@@ -154,6 +191,12 @@ func (p *Parser) GenerateKubernetesObjectReport() []client.Object {
 	return report
 }
 
+func (p *Parser) GetParsingErrors() []ParsingError {
```

Comment:
I'd prefer to see a comment on this func and perhaps I'd suggest renaming it as getters usually are considered to not mutate the state of the object they are called on, whereas here that's not the case.

Reply:
Yeah, sure: [e4c6a1b](https://github.com/Kong/kubernetes-ingress-controller/pull/3088/commits/e4c6a1be3aaee3a62c04e06695b8a5d7820eba40)